### PR TITLE
Add flag to "clean" command to target only messages with attachments.

### PR DESCRIPTION
**Pull request by jtwhale**
Adds a new "-attachments" switch to the clean command, which if provided limits the clean operation to messages with attachments

### CODE REVIEW

**Review by NaruDevnote — COMMENTED** [review]
Wouldn't `-a` be more fitting? All the clean flags are kept short, typing out `-clean 10 -attachments` is tedious.

**Comment by jtwhale** [comment]
Absolutely true, I'll shorten it

**Comment by jtwhale** [comment]
All done, gave it a quick test on my self-hosted instance

File: moderation/commands.go [hunk 1]
```diff
@@ -925,7 +933,17 @@ func AdvancedDeleteMessages(channelID int64, filterUser int64, regex string, toI
...
+		if attachmentFilterEnable {
+			if len(msgs[i].Attachments) > 0 {
+				toDelete = append(toDelete, msgs[i].ID)
+			} else {
+				continue
+			}
+		} else {
+                	toDelete = append(toDelete, msgs[i].ID)
+		}
+
```
Comment (DevKurka) [inline moderation/commands.go:946]: Why like that? Could easily stick to the current style by using the power of `continue`.

```go
// Check if attachment to ignore
if attachmentFilterEnable && len(msgs[i].Attachments) == 0 {
	continue
}
```

Comment (jtwhale) [reply moderation/commands.go]: @DevKurka

> 
> 
> Why like that? Could easily stick to the current style by using the power of `continue`.
> 
> ```go
> // Check if attachment to ignore
> if attachmentFilterEnable && len(msgs[i].Attachments) == 0 {
> 	continue
> }
> ```

You're 100% correct, I've amended it after testing.

**Pull request merged**